subxt/src/backend/chain_head/mod.rs

```diff
@@ -558,170 +588,230 @@ impl<T: Config + Send + Sync + 'static> Backend<T> for ChainHeadBackend<T> {
+                }
+            });
 
-                // When we get one, map it to the correct format (or for finalized ev, wait for the pinned block):
-                use subxt_rpcs::methods::chain_head::TransactionStatus as RpcTransactionStatus;
-                let tx_progress_ev = match tx_progress_ev {
-                    RpcTransactionStatus::Finalized { block } => {
-                        // We'll wait until we have seen this hash, to try to guarantee
-                        // that when we return this event, the corresponding block is
-                        // pinned and accessible.
-                        finalized_hash = Some(block.hash);
-                        continue;
-                    }
-                    RpcTransactionStatus::BestChainBlockIncluded { block: Some(block) } => {
-                        // Look up a pinned block ref if we can, else return a non-pinned
-                        // block that likely isn't accessible. We have no guarantee that a best
-                        // block on the node a tx was sent to will ever be known about on the
-                        // chainHead_follow subscription.
-                        let block_ref = match seen_blocks.get(&block.hash) {
-                            Some((_, block_ref)) => block_ref.clone().into(),
-                            None => BlockRef::from_hash(block.hash),
-                        };
-                        TransactionStatus::InBestBlock { hash: block_ref }
-                    }
-                    RpcTransactionStatus::BestChainBlockIncluded { block: None } => {
-                        TransactionStatus::NoLongerInBestBlock
-                    }
-                    RpcTransactionStatus::Broadcasted => TransactionStatus::Broadcasted,
-                    RpcTransactionStatus::Dropped { error, .. } => {
-                        TransactionStatus::Dropped { message: error }
-                    }
-                    RpcTransactionStatus::Error { error } => {
-                        TransactionStatus::Error { message: error }
-                    }
-                    RpcTransactionStatus::Invalid { error } => {
-                        TransactionStatus::Invalid { message: error }
-                    }
-                    RpcTransactionStatus::Validated => TransactionStatus::Validated,
-                };
-                return Poll::Ready(Some(Ok(tx_progress_ev)));
-            }
-        });
+            Ok(StreamOf(Box::pin(tx_stream)))
+        }
 
-        Ok(StreamOf(Box::pin(tx_stream)))
+        if self.submit_transactions_ignoring_follow_events {
+            submit_transaction_ignoring_follow_events(extrinsic, &self.methods).await
+        } else {
+            submit_transaction_tracking_follow_events::<T>(
+                extrinsic,
+                self.transaction_timeout_secs as u64,
+                &self.methods,
+                &self.follow_handle,
+            )
+            .await
+        }
```

Comment:
I just moved the existing logic into `submit_transaction_tracking_follow_events` and added an alternative path which is muc hsimpler for ignoring them